### ISSUE TITLE
CreateComputeInstance() shows “Not Supported”

### ISSUE THREAD

**Issue by ytaoeer** [NVIDIA/go-nvml#65]
go ： 1.18
go-nvml  v0.12.0-1
the code is below 
![image](https://github.com/NVIDIA/go-nvml/assets/30197785/ce61ef8a-7636-4061-8fe0-b31f551c32be)
![image](https://github.com/NVIDIA/go-nvml/assets/30197785/13505dbd-a93a-4109-9aaa-bb9a88cef3e1)
gpu instance can be created and using command line, i can create gi and ci.

**Comment by ytaoeer** [comment]
NVIDIA-SMI 535.54.03              
Driver Version: 535.54.03   
CUDA Version: 12.2
NVIDIA A100-PCIE-40GB

**Comment by elezar** [comment]
> gpu instance can be created and using command line, i can create gi and ci.

Which commands (I assume `nvidia-smi`) do you use?

**Comment by ytaoeer** [comment]
yes, "sudo nvidia-smi mig -i 0 -cgi 19 -C",i can create gi and ci；
but by using nvml go lib, i can just create gi, when using CreateComputeInstance() to create ci, it fail

**Comment by elezar** [comment]
Just as a sanity check. Do you have multiple A100 devices available? In the example you show above you access a device with index `1` wherease the `nvidia-smi` command accesses device `0`.

For what it's worth, we use the following flow to create a compute instance:
```
gi, ret = device.CreateGpuInstance(&giProfileInfo)
if ret != nvml.SUCCESS {
	return fmt.Errorf("error creating GPU instance: %v", ret)
}

ciProfileInfo, ret := gi.GetComputeInstanceProfileInfo(0, 0)
if ret != nvml.SUCCESS {
	return fmt.Errorf("error getting Compute instance profile info for: %v", ret)
}

_, ret = gi.CreateComputeInstance(&ciProfileInfo)
if ret != nvml.SUCCESS {
	return fmt.Errorf("error creating Compute instance: %v", ret)
}
```
in some of our toolking. Note the call to `GetComputeInstanceProfileInfo`. I believe you can use `0` for both

**Comment by ytaoeer** [comment]
> Just as a sanity check. Do you have multiple A100 devices available? In the example you show above you access a device with index `1` wherease the `nvidia-smi` command accesses device `0`.
> 
> For what it's worth, we use the following flow to create a compute instance:
> 
> ```
> gi, ret = device.CreateGpuInstance(&giProfileInfo)
> if ret != nvml.SUCCESS {
> 	return fmt.Errorf("error creating GPU instance: %v", ret)
> }
> 
> ciProfileInfo, ret := gi.GetComputeInstanceProfileInfo(0, 0)
> if ret != nvml.SUCCESS {
> 	return fmt.Errorf("error getting Compute instance profile info for: %v", ret)
> }
> 
> _, ret = gi.CreateComputeInstance(&ciProfileInfo)
> if ret != nvml.SUCCESS {
> 	return fmt.Errorf("error creating Compute instance: %v", ret)
> }
> ```
> 
> in some of our toolking. Note the call to `GetComputeInstanceProfileInfo`. I believe you can use `0` for both

it works. thank you very much. so i guess the problem is creating ComputeInstanceProfileInfo by myself. but according to the source code, the go lib only pass ComputeInstanceProfileInfo.Id to the nvml. 
![image](https://github.com/NVIDIA/go-nvml/assets/30197785/f3f34329-0bdf-48c0-ac9a-cbe81f4eaab8)

**Comment by elezar** [comment]
As far as I am aware, there are different versions of the `ComputeInstanceProfileInfo` struct -- or at least a versioned struct was introduced recently. It may be that constructing it manually created the wrong version and this is what caused it error.

**Comment by ytaoeer** [comment]
i use a release version go-nvml v0.12.0-1. the output is below
![image](https://github.com/NVIDIA/go-nvml/assets/30197785/5b727faf-cb47-40d8-aaf7-15536e391572)
![image](https://github.com/NVIDIA/go-nvml/assets/30197785/c0176d2a-3f60-4bdb-b7b7-294ad43cfee2)

**Comment by JunAr7112** [comment]
Hi @ytaoeer, let me know if you need any additional clarification, we should be able to create compute instances by using GetComputeInstanceProfileInfo(). For the setup, nvml.ComputeInstanceProfileInfo{} will directly create an empty instance of the compute instance struct itself. Will close this issue for now as GetComputeInstanceProfileInfo() can be used to fill the profile info.
